source/upgrade.txt

```diff
@@ -61,60 +61,60 @@ the version after v4.0 including any listed under v4.5.
-  these methods, see the :ref:`SocketSettings Example
-  <java-socketsettings-example>` in the Specify MongoClient Settings
-  guide.
-
-- This driver version removes the ``Filters.eqFull()`` method, which allowed you
-  to construct an equality filter when performing a vector search.
-  You can use the ``Filters.eq()`` method when instantiating a
-  ``VectorSearchOptions`` type, as shown in the following code:
-
-  .. code-block:: java
-     
-     VectorSearchOptions opts = vectorSearchOptions().filter(eq("x", 8));
-
-.. _java-breaking-changes-v5.0-observables:
-
-- This driver version removes the
-  ``org.mongodb.scala.ObservableImplicits.ToSingleObservableVoid`` implicit
-  class. This means the ``org.reactivestreams.Publisher[Void]`` type no longer
-  converts automatically to ``org.mongodb.scala.SingleObservable[Void]``. The also
-  API exposes ``org.mongodb.scala.Observable[Unit]`` instead of
-  ``org.mongodb.scala.Observable[Void]``. 
-  
-  .. After the 5.0 Scala API docs are released, this line will be uncommented. 
-    For more information, see the `Observable trait documentation <https://mongodb.github.io/mongo-java-driver/5.0/apidocs/mongo-scala-driver/org/mongodb/scala/Observable.html>`__.
+  - This driver version introduces the following changes to the ``ConnectionId`` class:
+    
+    - The ``ConnectionId`` constructor now accepts a value of type ``long`` as its second
+      parameter instead of type ``int``. Similarly, the constructor now accepts a value of
+      type ``Long`` as its third parameter instead of type ``Integer``. Because this change breaks
+      binary compatibility, recompile any existing code that calls the ``ConnectionId`` constructor.
+
+    - The ``withServerValue()`` method now accepts a parameter of type ``long`` rather than
+      type ``int``. This change breaks binary compatibility, so you must recompile any code
+      that calls the ``withServerValue()`` method.
+
+    - The ``getServerValue()`` method now returns a value of type ``Long`` instead of type
+      ``Integer``. Similarly, the ``getLocalValue()`` method returns a value of type
+      ``long`` instead of type ``int``. Because this change breaks both binary and source
+      compatibility, update any source code that uses these methods and rebuild your binary.
+
+  - This driver version removes the following record annotations:
+    
+    - ``BsonId``
+    - ``BsonProperty``
+    - ``BsonRepresentation``
+
+  - You must pass a timeout duration, which is the first parameter, to the
+    following ``SocketSettings`` builder methods as a ``long`` type:
+
+    - ``connectTimeout()``
+    - ``readTimeout()``
+
+    In earlier versions, this parameter is of type ``int`` for both methods. To view an example
+    that shows how to instantiate a ``SocketSettings`` instance by using
+    these methods, see the :ref:`SocketSettings Example
+    <java-socketsettings-example>` in the Specify MongoClient Settings
```

Comment:
1. The linked example does not show how to instantiate `SocketSettings`, nor is instantiation relevant here. I propose us to just say that the example shows how to call the updated methods.
2. The example uses `L` suffix, which is not needed there. Since it's neither a good practice, not a requirement to specify the suffix there, I propose us to remove the suffix.
3. Let's mention that the change breaks binary compatibility.